google/cloud/spanner_v1/database.py

```diff
@@ -443,6 +448,31 @@ def spanner_api(self):
             )
         return self._spanner_api
 
+    @property
+    def _channel_id(self):
+        """
+        Helper to retrieve the associated channelID for the spanner_api.
+        This property is paramount to x-goog-spanner-request-id.
+        """
+        with self.__transport_lock:
+            api = self.spanner_api
+            channel_id = self.__transports_to_channel_id.get(api._transport, None)
+            if channel_id is None:
+                channel_id = len(self.__transports_to_channel_id) + 1
+                self.__transports_to_channel_id[api._transport] = channel_id
+
+            return channel_id
```

Comment:
It is indeed cached as it is saved in the Database static __transports_to_channel_id map or would you prefer I name it with capital letters so that it is much easier to spot as __TRANSPORTS_TO_CHANNEL_ID?

Reply:
Yeah, what I meant is that I think we can just add an instance field `self.__channel_id` or something like that and store it there. Now it is indeed cached, but the cache look up includes taking a lock and then looking it up in a map. This is a function that is called for __every__ RPC, so keeping it as quick as possible is desirable, and I expect the extra memory usage to be negligible, as I don't think any application is going to create a very large number of database objects.

Reply:
Gotcha and thank you @olavloite, done!